### ISSUE TITLE
Add mapping to unelaborated types. UnelaboratedTypeOf(Decl)

### ISSUE THREAD

**Issue by GorNishanov** [GabrielDosReis/ipr#26]
```
typedef int MyInt;
MyInt* x;
```
The IPR declaration of x knows it is type int*, information about MyInt is discarded.
We would like to have a mapping that allows to discover what user has written to declare x.
Maybe something like:
```
Optional<Type> UnelaboratedTypeOf(const Decl&);
```
If an alias/typedef was involved, the UnelaboratedTypeOf(Decl) will give As_type(MyInt)* type.

**Comment by GabrielDosReis** [comment]
I agree that declarations, being generative, are places where additional declarative semantics fluff can be added.  However the type asked for here is still elaborated -- imagine uses of template aliases (one still wants elaboration).  There is a delicate balance to strike here between syntax and semantics.  Ideally there should be a separate syntax tree that is mapped to the semantics (IPR), and there should be (ideally) a way to query the original alias type used to declare a name in the IPR decl node.
